sdk/src/azure/core/az_span.c

```diff
@@ -127,7 +124,12 @@ static bool _is_valid_start_of_number(uint8_t first_byte)
 #pragma warning(disable : 4996)
 #endif
 
-static void _az_span_ato_number_helper(az_span source, char* format, void* result, bool* success)
+static void _az_span_ato_number_helper(
+    az_span source,
+    bool is_negative_allowed,
```

Comment:
Use a enum value to represent what true/false.  It makes the code much more readable.

Reply:
Can you expand on what you mean by that? Is the code more readable at the call site?

Do you mean something like this?
```C
enum
{
  _az_NEGATIVE_SIGN_NOT_ALLOWED = 0,
  _az_NEGATIVE_SIGN_ALLOWED = 1,
````

Reply:
In the interest of time, and since this is regarding readability of implementation detail, that we can fix later, I am going to merge this PR and address this separately. Please let me know what change you would like me to make with the enum, @RickWinter